packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -830,6 +831,13 @@ - (void)startVideoRecordingWithCompletion:(void (^)(FlutterError *_Nullable))com
                                      details:nil]);
       return;
     }
+    // do not call startWriting in didOutputSampleBuffer to prevent state in which
```

Comment:
nit: "do not move startWriting call to didOutputSampleBuffer... "

Because there should only be 1 place to call `startWriting`.